eslint-plugin should not import from typescript-estree

**Repro**
`@typescript-eslint/eslint-plugin` should not import from `@typescript-eslint/typescript-estree`, since it does not have a direct dependency on this package.  Instead, it should only import from `@typescript-eslint/experimental-utils`.

This issue is hidden when using a package manager which flattens dependencies like `npm`, since `@typescript-eslint/parser` does have a dependency on `@typescript-eslint/typescript-estree`, and `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` are typically used together.

However, when using a package manager like `pnpm` which [does not flatten dependencies](https://pnpm.js.org/docs/en/faq.html#pnpm-does-not-work-with-your-project-here), `@typescript-eslint/eslint-plugin` fails with the following error:

```
Error: Cannot find module '@typescript-eslint/typescript-estree'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:690:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (D:\Git\typescript-eslint-test\node_modules\.registry.npmjs.org\@typescript-eslint\eslint-plugin\1.12.0_db854cf46887ef4aa7b9323cccc417a5\node_modules\@typescript-eslint\eslint-plugin\dist\rules\prefer-readonly.js:17:29)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
```

Repro steps:

```
git clone https://github.com/mikeharder/typescript-eslint-test

npm install
npm run lint (works fine)

git clean -xdf
npm install -g pnpm
pnpm install
npm run lint (fails)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.12.0` |
| `@typescript-eslint/parser`        | `1.12.0` |
| `TypeScript`                       | `3.5.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |
| `pnpm`                              | `3.5.7` |

--- a/packages/eslint-plugin/src/rules/prefer-readonly.ts
+++ b/packages/eslint-plugin/src/rules/prefer-readonly.ts
@@ -2,7 +2,10 @@ import * as tsutils from 'tsutils';
 import ts from 'typescript';
 import * as util from '../util';
 import { typeIsOrHasBaseType } from '../util';
-import { TSESTree, AST_NODE_TYPES } from '@typescript-eslint/typescript-estree';
+import {
+  TSESTree,
+  AST_NODE_TYPES,
+} from '@typescript-eslint/experimental-utils';
 
 type MessageIds = 'preferReadonly';
 

--- a/packages/eslint-plugin/src/rules/prefer-regexp-exec.ts
+++ b/packages/eslint-plugin/src/rules/prefer-regexp-exec.ts
@@ -1,4 +1,4 @@
-import { TSESTree } from '@typescript-eslint/typescript-estree';
+import { TSESTree } from '@typescript-eslint/experimental-utils';
 import { createRule, getParserServices, getTypeName } from '../util';
 import { getStaticValue } from 'eslint-utils';
 

--- a/packages/eslint-plugin/src/rules/triple-slash-reference.ts
+++ b/packages/eslint-plugin/src/rules/triple-slash-reference.ts
@@ -1,10 +1,5 @@
 import * as util from '../util';
-import {
-  Literal,
-  Node,
-  TSExternalModuleReference,
-} from '@typescript-eslint/typescript-estree/dist/ts-estree/ts-estree';
-import { TSESTree } from '@typescript-eslint/typescript-estree';
+import { TSESTree } from '@typescript-eslint/experimental-utils';
 
 type Options = [
   {
@@ -55,14 +50,14 @@ export default util.createRule<Options, MessageIds>({
     },
   ],
   create(context, [{ lib, path, types }]) {
-    let programNode: Node;
+    let programNode: TSESTree.Node;
     const sourceCode = context.getSourceCode();
     const references: ({
       comment: TSESTree.Comment;
       importName: string;
     })[] = [];
 
-    function hasMatchingReference(source: Literal) {
+    function hasMatchingReference(source: TSESTree.Literal) {
       references.forEach(reference => {
         if (reference.importName === source.value) {
           context.report({
@@ -78,14 +73,14 @@ export default util.createRule<Options, MessageIds>({
     return {
       ImportDeclaration(node) {
         if (programNode) {
-          const source = node.source as Literal;
+          const source = node.source as TSESTree.Literal;
           hasMatchingReference(source);
         }
       },
       TSImportEqualsDeclaration(node) {
         if (programNode) {
-          const source = (node.moduleReference as TSExternalModuleReference)
-            .expression as Literal;
+          const source = (node.moduleReference as TSESTree.TSExternalModuleReference)
+            .expression as TSESTree.Literal;
           hasMatchingReference(source);
         }
       },

